Modules/Devices/src/Devices.ConsumerApi/Controllers/IdentitiesController.cs

```diff
@@ -76,6 +77,15 @@ public async Task<IActionResult> ApproveDeletionProcess([FromRoute] string id, C
         var response = await _mediator.Send(new ApproveDeletionProcessCommand(id), cancellationToken);
         return Ok(response);
     }
+
+    [HttpGet("Self/DeletionProcesses/{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesError(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetDeletionProcess([FromRoute] string id, CancellationToken cancellationToken)
+    {
+        var response = await _mediator.Send(new GetDeletionProcessQuery { Id = id }, cancellationToken);
+        return Ok(response);
+    }
```

Comment:
Maybe you would also need integration tests for this but I suppose we should wait for decision of @tnotheis on this one.